composer/datasets/in_context_learning_evaluation.py

```diff
@@ -85,582 +109,764 @@ def _make_padded_input(context_enc, continuation_enc, max_seq_len, pad_tok_id, p
+                'fewshot_rng': fewshot_rng,
+            },
+        )
 
-    def _format_prompt_and_fewshot(self, num_fewshot: int, prompt_string: str, example_delimiter: str,
-                                   continuation_delimiter: str, question_prelimiter: str, cot_delimiter: str,
-                                   fewshot_rng: random.Random, sample_idx: int) -> str:
-        """Formats the prompt fewshot examples for test sample `sample_idx`.
+    def __getitem__(self, index: int) -> Dict:
+        return self.dataset[index]
+
+    def __len__(self) -> int:
+        return len(self.dataset)
+
+    def get_num_samples_in_batch(self, batch: Dict) -> int:
+        return batch['input_ids'].shape[0]
 
-        Randomly select `num_fewshot` samples from the dataset (not including the sample at `sample_idx`) and format
-        them each as follows `{example_delimiter}{question_prelimiter}{context}{continuation_delimiter}{chain_of_thought}{cot_delimiter}{answer}`.
+    def check_defaults_are_set(self, dict_of_defaults: dict) -> None:
+        if all(v for v in dict_of_defaults.values()):
+            return
+        raise ValueError(
+            f"{type(self).__name__} missing required variable(s): {', '.join([k for k, v in dict_of_defaults.items() if not v])}"
+        )
 
-        `chain_of_thought` will default to empty if not present in the dataset but `context` and `answer` must be present.
+    def _read_dataset(self,
+                      dataset_uri: str,
+                      destination_path: str,
+                      hf_loading_vars: Dict = None,
+                      hf_parsing_map: Dict = None) -> transformers.Dataset:
+        """
+        Reads a dataset and handles parsing it from HuggingFace.
+        Args:
+            dataset_uri (str): A local path, a remote path beginning with ``s3://`` or another backend, or a HuggingFace dataset uri.
+                Alternate backends must be supported by :meth:`composer.utils.maybe_create_object_store_from_uri`.
+            destination_path (str): A local path where the data will be stored
+            hf_loading_vars (Dict): If parsing from HuggingFace, keyword args that will be passed into load_dataset
+            hf_parsing_map (Dict): Dictionary in the form of {icl_key: [hf_col1, hf_col2]} that will map one or more hf columns, in order, to ICL dataset columns
 
-        Returns the formatted prompt_string + concatenated list of formatted few shot examples.
+        Returns:
+            dataset: a loaded HF dataset
         """
-        prompt_and_fewshot = prompt_string
+        try:
+            from datasets import load_dataset  # pyright: ignore [reportGeneralTypeIssues]
+        except ImportError as e:
+            raise MissingConditionalImportError(
+                extra_deps_group='nlp',
+                conda_package='datasets',
+                conda_channel='conda-forge',
+            ) from e
+        if _check_if_huggingface_uri(dataset_uri):
```

Comment:
See my other comment about prefixing hugging face URIs to indicate they are not local paths